### PR TITLE
Add bool type for attribute and use it in dropout_op.

### LINKED ISSUE #4215: Add bool type for paddle::framework::Attribute
Even though we can use `int` type to express `bool` type. But if use `int` type, some checking code in operators is as follow:

```cpp
    PADDLE_ENFORCE(ctx.Attr<int>("is_training") == 0 ||
                   ctx.Attr<int>("is_training") == 1);
```

And the bool type attributes are used frequently in operators.  So, I think it's necessary to add bool type.

### PR DESCRIPTION
**Pull request by qingqing01**
Fix #4215

### CODE REVIEW

**Comment by lcy-seso** [comment]
这里 https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/cross_entropy_op.cc#L39 也需要改掉。不过可以提新的PR。

**Review by reyoung — CHANGES_REQUESTED** [review]
I do not think bool is needed because it could be simulated by `int`.

File: paddle/operators/dropout_op.cc [hunk 1]
```diff
@@ -29,13 +29,10 @@ class DropoutOp : public framework::OperatorWithKernel {
     PADDLE_ENFORCE_NOT_NULL(ctx.InputVar("X"), "Input(X) must not be null.");
     PADDLE_ENFORCE_GE(ctx.Attr<float>("dropout_prob"), 0);
     PADDLE_ENFORCE_LE(ctx.Attr<float>("dropout_prob"), 1);
-    // TODO(xinghai-sun): remove this check after swtiching to bool
-    PADDLE_ENFORCE(ctx.Attr<int>("is_training") == 0 ||
-                   ctx.Attr<int>("is_training") == 1);
 
     auto dims = ctx.Input<Tensor>("X")->dims();
     ctx.Output<LoDTensor>("Out")->Resize(dims);
-    if (ctx.Attr<int>("is_training") == 1) {
```
Comment (reyoung) [inline paddle/operators/dropout_op.cc]: Actually, zero is false, and non-zero is true.

File: paddle/operators/dropout_op.cc [hunk 2]
```diff
@@ -49,8 +46,7 @@ class DropoutOpMaker : public framework::OpProtoAndCheckerMaker {
       : OpProtoAndCheckerMaker(proto, op_checker) {
     AddAttr<AttrType>("dropout_prob", "Probability of setting units to zero.")
         .SetDefault(.5f);
-    // TODO(xinghai-sun): use bool for is_training after bool is supported.
-    AddAttr<int>("is_training", "Whether in training phase.").SetDefault(1);
```
Comment (reyoung) [inline paddle/operators/dropout_op.cc]: SetDefault(true)

**Comment by reyoung** [comment]
Please reference #4239

**Review by reyoung — DISMISSED** [review]

**Review by lcy-seso — APPROVED** [review]
LGTM, thank you.

**Pull request merged**